includes/sanitizers/class-amp-tag-and-attribute-sanitizer.php

```diff
@@ -138,24 +138,18 @@ private function process_node( $node ) {
 				// This should not happen very often, but...
 				// If we're here, then we're not sure which spec should
 				// be used. Let's use the top scoring ones.
-				foreach( $spec_ids_sorted as $id ) {
-					$attr_spec_list = array_merge( $attr_spec_list, $rule_spec_list_to_validate[ $id ][AMP_Rule_Spec::ATTR_SPEC_LIST] );
+				foreach ( $spec_ids_sorted as $id ) {
+					$spec_list = isset( $rule_spec_list_to_validate[ $id ][ AMP_Rule_Spec::ATTR_SPEC_LIST ] ) ? $rule_spec_list_to_validate[ $id ][ AMP_Rule_Spec::ATTR_SPEC_LIST ] : null;
+					if ( ! $this->is_missing_mandatory_attribute( $spec_list, $node ) ) {
+						$attr_spec_list = array_merge( $attr_spec_list, $spec_list );
+					}
 				}
 			}
-		}
-
-		// If an attribute is mandatory and we don't have it, just remove the node and move on.
-		foreach ( $attr_spec_list as $attr_name => $attr_spec_rule_value ) {
-			$is_mandatory =
-				isset( $attr_spec_rule_value[ AMP_Rule_Spec::MANDATORY ] )
-				? (bool) $attr_spec_rule_value[ AMP_Rule_Spec::MANDATORY ]
-				: false;
-			$attribute_exists = $node->hasAttribute( $attr_name );
+		} // End if().
 
-			if ( $is_mandatory && ! $attribute_exists ) {
-				$this->remove_node( $node );
-				return;
-			}
```

Comment:
This is mainly decomposed into the function below, `is_missing_mandatory_attribute()`.